p8e-sdk/src/main/kotlin/io/p8e/proxy/PermissionUpdater.kt

```diff
@@ -17,12 +18,16 @@ class PermissionUpdater(
 ) {
 
     fun saveConstructorArguments() {
+        // TODO this can be optimized by checking the recitals and record groups and determining what subset, if any,
+        // of input facts need to be fetched and stored in order only save the objects that are needed by some of
+        // the recitals
         contract.inputsList.threadedMap(executor) { fact ->
-            val message = contractManager.loadProto(
-                fact.dataLocation.ref.hash,
-                fact.dataLocation.classname
-            )
-            contractManager.saveProto(message, audience = audience)
+            with (contractManager.client) {
+                val obj = this.loadObject(fact.dataLocation.ref.hash)
+                val inputStream = ByteArrayInputStream(obj)
```

Comment:
we dont care about classname?

Reply:
All the classname does it cast the loaded proto. In this case we are now loading the bytes and saving the bytes directly so we never need to cast to the classname.